oio/cli/election/election.py

```diff
@@ -0,0 +1,126 @@
+# Copyright (C) 2017 OpenIO SAS, as part of OpenIO SDS
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from logging import getLogger
+from cliff import lister, show
+
+
+def format_json(parsed_args, json):
+    from oio.common.json import json as jsonlib
+
+    if json and parsed_args.formatter == 'table':
+        json = jsonlib.dumps(json, indent=4, sort_keys=True)
+    return json
+
+
+class PingElection(lister.Lister):
+    """Trigger or refresh an election"""
+
+    log = getLogger(__name__ + '.PingElection')
+
+    def get_parser(self, prog_name):
```

Comment:
All commands of this class take the same parameters. You should consider creating a function (or a mixin class, like in [`oio/cli/object/object.py`](https://github.com/open-io/oio-sds/blob/4.x/oio/cli/object/object.py#L23)) to create the parser. By the way, `reference` is not mandatory, you can interact with an election knowing only the CID.